agent/php_api.c

```diff
@@ -177,6 +181,23 @@ PHP_FUNCTION(newrelic_notice_error) {
     nr_free(buf);
     nr_free(stack_json);
 
+    RETURN_TRUE;
+  }
+  if (determine) {
+    char* buf1 = nr_strndup(errormsgstr, errormsglen);
+    char* buf2 = nr_strndup(error_file, error_file_len);
+    char* buf3 = nr_strndup(error_context, error_context_len);
+    char* stack_json = nr_php_backtrace_to_json(NULL TSRMLS_CC);
+
+    nr_txn_record_error_with_additional_attributes(
+        NRPRG(txn), priority, true, buf1, errclass, buf2,
+        error_line, buf3, error_number, stack_json);
+
+    nr_free(buf1);
+    nr_free(buf2);
+    nr_free(buf3);
+    nr_free(stack_json);
```

Comment:
What is the purpose of copying these strings only to immediately free them? Is there a reason why we cannot just directly pass in `errormsgstr`, `error_file`, and `error_context`?

Reply:
Good point, fixed: [b8e6402](https://github.com/newrelic/newrelic-php-agent/pull/942/commits/b8e64022e5b029a057aece820b663e4b444e556b)